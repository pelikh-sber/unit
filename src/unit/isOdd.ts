export const isOdd = (value: any) => {
    if (typeof value === 'number') {
        return value % 2 === 0
    }
    throw Error('Это не число')
}