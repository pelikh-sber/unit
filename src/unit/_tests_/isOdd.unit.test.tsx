import { isOdd } from "../isOdd";

describe("Функциональность isOdd", () => {
  it("работает успешно при ожидаемом параметре", () => {
    expect(isOdd(1)).toBe(false);
  });
  it("работает успешно при неожидаемом типе параметра", () => {
    expect(() => isOdd("")).toThrowErrorMatchingSnapshot();
  });
});
