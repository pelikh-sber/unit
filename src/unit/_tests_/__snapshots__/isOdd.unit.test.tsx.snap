// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Функциональность isOdd работает успешно при неожидаемом типе параметра 1`] = `"Это не число"`;
